boards/arm/nrf5340_dk_nrf5340/doc/index.rst

```diff
@@ -0,0 +1,260 @@
+.. _nrf5340_dk_nrf5340:
+
+nRF5340-DK
+##########
+
+Overview
+********
+
+The nRF5340 DK is a single-board development kit for evaluation
+and development on the Nordic nRF5340 System-on-Chip (SoC).
+
+The nRF5340 is a dual-core SoC based on the Arm® Cortex®-M33 architecture, with:
+
+* a full-featured ARM Cortex-M33F core with DSP instructions, FPU, and
+  ARMv8-M Security Extension, running at up to 128 MHz, referred to as
+  the **Application MCU**
+* a secondary ARM Cortex-M33 core, with a reduced feature set, running at
+  a fixed 64 MHz, referred to as the **Network MCU**.
+
+The nrf5340_dk_nrf5340_cpuapp provides support for the Application MCU on
```

Comment:
Given @carlescufi question about the Application and Network **MCU** name, it's amusing that the board name has **cpu** in it (instead of **mcu**). Oh well.

Reply:
Yes, I dislike that but it's for consistency with other existing multi-core boards.